tests/test_trainer.py

```diff
@@ -0,0 +1,69 @@
+import shutil
+import unittest
+from pathlib import Path
+
+from test_utils import TEST_CONFIG
+
+from trainlib.trainer import SegmentationTrainer
+
+
+class TestSegmentationTrainer(unittest.TestCase):
+    config = TEST_CONFIG
+
+    def test_in_order(self):
+        self._test_one_epoch()
+        self._test_folder_structure()
+
+    def _test_one_epoch(self):
+        trainer = SegmentationTrainer(config=self.config)
+        trainer.run()
+
+    def _test_folder_structure(self):
+        """Check that all folders for logging have been created"""
+        # check runs were created
+        self.assertPathExists("runs")
+        # check subdirs of runs (logs, output, trainlib, patch) were created
+        self.assertPathExists("runs/name/logs")
+        self.assertPathExists("runs/name/logs/train_logs.csv", is_file=True)
+
+        self.assertPathExists("runs/name/output")
+        self.assertPathExists("runs/name/output/preds")
+        self.assertPathExists("runs/name/output/preds/image.pt", is_file=True)
+        self.assertPathExists("runs/name/output/preds/label.pt", is_file=True)
+        self.assertPathExists("runs/name/output/preds/pred_epoch_1.pt", is_file=True)
+        for file in [
+            "emissions.csv",
+            "HausdorffDistance_raw.csv",
+            "MeanDice_raw.csv",
+            "metrics.csv",
+            "SurfaceDistance_raw.csv",
+            "val_mean_dice_raw.csv",
+        ]:
+            self.assertPathExists(f"runs/name/output/{file}", is_file=True)
+
+        self.assertPathExists("runs/name/trainlib")
+        # check config and requirements yaml were created
+        self.assertPathExists("runs/name/config.yaml", is_file=True)
+        self.assertPathExists("runs/name/requirements.txt", is_file=True)
+
+        # check model dir was created
+        self.assertPathExists("models")
+        # check model dir is not empty
+        self.assertIsNotNone(next(Path("models").iterdir(), None))
+
+        # cleanup
+        shutil.rmtree(self.config.run_id.split("/")[0])
+        shutil.rmtree(self.config.model_dir)
+        shutil.rmtree(self.config.data.cache_dir)
+
+    def assertPathExists(self, fn: str, is_file: bool = False):  # noqa N802
```

Comment:
Why do we overwrite `assertPathExists` (from the base class)? 
Maybe add a comment/docstring to explain

Reply:
I tried to find this function, but wasn't able to. So I've created it. Will revert now.

Reply:
Sorry, can't find it, so I was probably mistaken.

Still wondering if we should use snake_case: `assertPathExists` -> `assert_path_exists`

Reply:
I tried to copy the unittest style, which does not use `snake_case`. What could be considered as well is making a TestCase subclass, that provides additional features. Will do so, if `assertPathExists` (or another function) is needed again in the future.